### PR TITLE
feat: [CHAOS-2098]: Network map creation graph with route fixes and idb setup

### PR DESCRIPTION
**Pull request by arkajyotiMukherjee**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by SahilKr24 — APPROVED** [review]

**Review by rajarshee-harness — APPROVED** [review]

**Review by abhinavrastogi-harness — APPROVED** [review]
Let's make sure CD team is aware of the IDP version upgrade.
cc @srdjan-arsic @kalyansai-harness

File: src/modules/10-common/RouteDefinitions.ts [hunk 1]
```diff
@@ -457,8 +468,19 @@ const routes = {
...
+    ({
+      accountId: _accountId,
+      orgIdentifier,
+      projectIdentifier,
+      module,
+      dAgentId,
+      networkMapId,
+      ...rest
+    }: Partial<ProjectPathProps & ModulePathParams & NetworkMapPathProps & NetworkMapQueryParams>) => {
+      const queryString = qs.stringify(rest, { skipNulls: true })
```
[the remark this answers is not in the document]
Comment (arkajyotiMukherjee) [reply src/modules/10-common/RouteDefinitions.ts:480]: This will be addressed in the very next PR. You have my word 🙏🏼

**Review by chetanvekariya-harness — APPROVED** [review]

File: src/modules/73-discovery/components/NetworkGraph/__tests__/mocks.ts [hunk 2]
```diff
@@ -5,6 +5,7 @@
  * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
  */
 
+/* istanbul ignore file */
```
Comment (pankajsingh-harness) [inline src/modules/73-discovery/components/NetworkGraph/__tests__/mocks.ts:8]: Is this expected here?

File: src/modules/73-discovery/hooks/useDiscoveryIndexedDBHook.ts [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+}
+
+export const DiscoveryIndexedDBPrimaryKeys = {
+  NETWORK_MAP: 'identity'
+}
+
+const logger = loggerFor(ModuleName.COMMON)
+
+interface DiscoveryObjectStore {
+  name: 'networkMap'
```
Comment (pankajsingh-harness) [inline src/modules/73-discovery/hooks/useDiscoveryIndexedDBHook.ts:27]:
```suggestion
  name: 'DiscoveryObjectStoreNames.NETWORK_MAP'
```

File: src/modules/73-discovery/components/NetworkGraph/__tests__/mocks.ts (hunk 2 above)
Comment (arkajyotiMukherjee) [reply src/modules/73-discovery/components/NetworkGraph/__tests__/mocks.ts:8]: yes this is just mock data for tests. somehow it was coming up in coverage.

File: src/modules/73-discovery/hooks/useDiscoveryIndexedDBHook.ts (hunk 3 above)
Comment (arkajyotiMukherjee) [reply src/modules/73-discovery/hooks/useDiscoveryIndexedDBHook.ts:27]: Missed this. the type would be just DiscoveryObjectStoreNames enum. I'll make the change in the next PR so as to not dismiss the approvals. Thanks for the review.

**Pull request merged**